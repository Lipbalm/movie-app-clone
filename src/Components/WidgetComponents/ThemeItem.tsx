import React, { FC } from "react";
import {
  ThemeItemWrapper,
  ThumnailWrapper,
  Thumnail,
  ThemeItemLayer,
  OverlayTitleTag
} from "./WidgetStyle";

export const BigItem: FC = () => {
  return (
    <ThemeItemWrapper margin={"0 0 0 0"}>
      <ThumnailWrapper width={"36.75rem"} height={"24rem"}>
        <Thumnail />
      </ThumnailWrapper>
      <ThemeItemLayer>
        <OverlayTitleTag />
      </ThemeItemLayer>
    </ThemeItemWrapper>
  );
};

export const TinyItem: FC = () => {
  return <></>;
};

Object.entries({ width: "100px", height: "100px" }).reduce((a, c) => {
  return (a += `${c[0]}: ${c[1]};`);
}, ``);
