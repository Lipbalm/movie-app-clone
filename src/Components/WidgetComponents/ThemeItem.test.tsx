import React from "react";
import { render } from "@testing-library/react";
import { BigItem, TinyItem } from "./ThemeItem";

const itemSetup = (sep: string) => {
  return sep === "B" ? { ...render(<BigItem />) } : { ...render(<TinyItem />) };
};

describe("<BigItem/> & <TinyItem/>", () => {
  it("snapshot", () => {
    const { container } = itemSetup("B");
    expect(container).toMatchSnapshot();
  });

  it("display Item", () => {
    const { getByTestId } = itemSetup("B");

    const item = getByTestId("item");
    const thumnailWrapper = getByTestId("thumnailWrapper");
    const thumnail = getByTestId("thumnail");
    const layer = getByTestId("layer");
    const link = getByTestId("link");

    expect(item).toBeVisible();
    expect(thumnailWrapper).toBeVisible();
    expect(thumnail).toBeVisible();
    expect(layer).toBeVisible();
    expect(link).toBeVisible();
  });
});
