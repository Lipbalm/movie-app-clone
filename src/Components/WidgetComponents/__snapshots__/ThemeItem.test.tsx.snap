// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigItem/> & <TinyItem/> snapshot 1`] = `<div />`;
